### PR TITLE
fix: Add revisionHistoryLimit override to cluster-autoscaler

### PR DESCRIPTION
**Pull request by bodgit**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:

Allows overriding the `revisionHistoryLimit` on the cluster autoscaler deployment. The default value is 10 which is the implicit Kubernetes default.

#### Which issue(s) this PR fixes:

#5701 was previously opened to add this but the submitter cancelled the PR.

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by vadasambar** [comment]
image

**Comment by vadasambar** [comment]
Thank you for the PR @bodgit 
/lgtm

**Comment by vadasambar** [comment]
@mwielgus , @gjtempleton need approval.

**Comment by Shubham82** [comment]
@bodgit please resolve the merge conflicts, so that the failed test will be passed.

**Comment by bodgit** [comment]
@Shubham82 done

**Comment by Shubham82** [comment]
@bodgit you need to bump the chart version so that the failed test case will be passed.

**Comment by bodgit** [comment]
> @bodgit you need to bump the chart version so that the failed test case will be passed.

Sorry, second attempt. Should be fixed now.

**Comment by Shubham82** [comment]
> > @bodgit you need to bump the chart version so that the failed test case will be passed.
> 
> Sorry, second attempt. Should be fixed now.

Don't be sorry for it.

File: charts/cluster-autoscaler/Chart.yaml [hunk 1]
```diff
@@ -11,4 +11,4 @@ name: cluster-autoscaler
 sources:
   - https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler
 type: application
-version: 9.31.0
+version: 9.31.1
```
Comment (gjtempleton) [inline charts/cluster-autoscaler/Chart.yaml:14]: Adding a feature, so please make this a minor version bump

Comment (bodgit) [reply charts/cluster-autoscaler/Chart.yaml]: Done

**Comment by gjtempleton** [comment]
Thanks!
/lgtm
/approve

**Pull request merged**